### PR TITLE
LPS-187587 - Add Autom. Test UntoggledSettingsAreAppliedImmediately

### PR DESCRIPTION
**Pull request by eltonccdantas**
[Ticket](https://liferay.atlassian.net/browse/LPS-187587)

Given home page
And underline class is applied to Powered by Liferay
// verify show underline config in accessibility menu is on
// verify link element "Powered by Liferay" does have "c-prefers-link-underline" class present
When access Accessibility Menu
And toggle OFF show underline effect in links 
Then Powered by Liferay link does not have underline class applied
// assert link element "Powered by Liferay" does not have "c-prefers-link-underline" class present

### CODE REVIEW

**Comment by john-co** [comment]
I pushed in a commit to refactor the task steps. Please remember to not include `//` comments as part of task steps. They are only to provide hints on what specific poshi actions may be included in the actual Given When Then steps.

File: modules/apps/accessibility/accessibility-test/src/testFunctional/tests/AccessibilityMenu.testcase [hunk 1]
```diff
@@ -272,24 +274,41 @@ definition {
...
+				locator1 = "ToggleSwitch#ANY_TOGGLE_SWITCH");
+		}
 
-		//}
+		task ("And underline class is applied to body element") {
+			AssertChecked.assertCheckedNotVisible(
+				key_toggleSwitchLabel = "Underlined Links",
+				locator1 = "ToggleSwitch#ANY_TOGGLE_SWITCH");
 
+			AssertHTMLSourceTextPresent(value1 = "c-prefers-link-underline");
```
Comment (john-co) [inline modules/apps/accessibility/accessibility-test/src/testFunctional/tests/AccessibilityMenu.testcase:294]: Please target the body class specifically with `AssertElementPresent(`
locator is something like: `//body(contains...c-prefers-link-underline...)`

We want to make sure the class is applied globally on the page so we want to verify that the class is at the body element level. Not just anywhere in the html source.

File: modules/apps/accessibility/accessibility-test/src/testFunctional/tests/AccessibilityMenu.testcase [hunk 2]
```diff
@@ -272,24 +274,41 @@ definition {
...
+		}
+
+		task ("And toggle OFF show underline effect in links") {
+			Uncheck.uncheckToggleSwitch(
+				key_toggleSwitchLabel = "Underlined Links",
+				locator1 = "ToggleSwitch#ANY_TOGGLE_SWITCH");
+		}
+
+		task ("Then body element does not have underline class applied") {
+			AssertHTMLSourceTextNotPresent(value1 = "c-prefers-link-underline");
```
Comment (john-co) [inline modules/apps/accessibility/accessibility-test/src/testFunctional/tests/AccessibilityMenu.testcase:310]: same as above comment

**Review by john-co — CHANGES_REQUESTED** [review]
Minor edits requested.

File: modules/apps/accessibility/accessibility-test/src/testFunctional/tests/AccessibilityMenu.testcase (hunk 1 above)
Comment (eltonccdantas) [reply modules/apps/accessibility/accessibility-test/src/testFunctional/tests/AccessibilityMenu.testcase:294]: I tried `AssertHTMLSourceTextPresent(value1 = "//body[contains(@class, 'c-prefers-link-underline')]");` 
But it fails... 😢

Comment (eltonccdantas) [reply modules/apps/accessibility/accessibility-test/src/testFunctional/tests/AccessibilityMenu.testcase:294]: Also tried other options like `	AssertHTMLSourceTextPresent.assertHTMLSourceTextPresentNoSPARefresh(value1 = "c-prefers-link-underline");` and got the same fail

**Comment by eltonccdantas** [comment]
liferay/ci:test:accessibility-menu

**Comment by eltonccdantas** [comment]
ci:test:accessibility-menu:nocomile

File: modules/apps/accessibility/accessibility-test/src/testFunctional/tests/AccessibilityMenu.testcase (hunk 1 above)
Comment (john-co) [reply modules/apps/accessibility/accessibility-test/src/testFunctional/tests/AccessibilityMenu.testcase]: What about just `AssertElementPresent`?

**Comment by eltonccdantas** [comment]
🚀 

image

File: modules/apps/accessibility/accessibility-test/src/testFunctional/tests/AccessibilityMenu.testcase (hunk 1 above)
Comment (eltonccdantas) [reply modules/apps/accessibility/accessibility-test/src/testFunctional/tests/AccessibilityMenu.testcase]: I thought `AssertElementPresent` was not used for HTML actually...

Comment (eltonccdantas) [reply modules/apps/accessibility/accessibility-test/src/testFunctional/tests/AccessibilityMenu.testcase]: Aright, done!

File: modules/apps/accessibility/accessibility-test/src/testFunctional/tests/AccessibilityMenu.testcase (hunk 2 above)
Comment (eltonccdantas) [reply modules/apps/accessibility/accessibility-test/src/testFunctional/tests/AccessibilityMenu.testcase]: Done ✅

**Comment by eltonccdantas** [comment]
ci:test:sf

**Comment by eltonccdantas** [comment]
ci:test:accessibility-menu:nocompile

File: modules/apps/accessibility/accessibility-test/src/testFunctional/tests/AccessibilityMenu.testcase [hunk 3]
```diff
@@ -272,24 +274,51 @@ definition {
...
+				locator1 = "ToggleSwitch#ANY_TOGGLE_SWITCH");
+		}
 
+		task ("verify body element does have 'c-prefers-link-underline' class present") {
+			var key_class = ${key_class};
+
+			AssertElementPresent(
+				key_class = "c-prefers-link-underline",
+				locator1 = "AccessibilityMenu#SELECTED_CLASS");
+		}
```
Comment (john-co) [inline modules/apps/accessibility/accessibility-test/src/testFunctional/tests/AccessibilityMenu.testcase:301]: Keep the poshi steps, remove the task descriptions. When we create the test scenarios please don't add in the comments. The comments are just clues for testers in translating the Given When Then steps to poshi. Please only use Given When Then format for structuring tasks.

File: modules/apps/accessibility/accessibility-test/src/testFunctional/paths/AccessibilityMenu.path [hunk 4]
```diff
@@ -21,6 +21,11 @@
 	<td>//*[@class="modal-title"][contains(text(),"${key_modal_title}")]</td>
 	<td></td>
 </tr>
+<tr>
+	<td>SELECTED_CLASS</td>
```
Comment (john-co) [inline modules/apps/accessibility/accessibility-test/src/testFunctional/paths/AccessibilityMenu.path:25]: Rename to BODY_SPECIFIC_CLASS

**Comment by eltonccdantas** [comment]
ci:test:sf

File: modules/apps/accessibility/accessibility-test/src/testFunctional/paths/AccessibilityMenu.path (hunk 4 above)
Comment (eltonccdantas) [reply modules/apps/accessibility/accessibility-test/src/testFunctional/paths/AccessibilityMenu.path]: Alrighty

**Comment by eltonccdantas** [comment]
ci:test:sf

**Comment by john-co** [comment]
Just started reviewing... :hourglass_flowing_sand:

File: modules/apps/accessibility/accessibility-test/src/testFunctional/tests/AccessibilityMenu.testcase (hunk 3 above)
Comment (john-co) [reply modules/apps/accessibility/accessibility-test/src/testFunctional/tests/AccessibilityMenu.testcase]: remember to do this

**Comment by john-co** [comment]
https://github.com/liferay-peds/liferay-portal/pull/173#discussion_r1275488635

**Comment by john-co** [comment]
ci:test:accessibility-menu:nocompile

**Comment by john-co** [comment]
ci:test:sf

**Review by john-co — APPROVED** [review]
Please see adjustments

**Comment by john-co** [comment]
ci:forward

**Pull request closed without merging**